src/ConfigGen/Program.cs

```diff
@@ -418,26 +418,51 @@ static string createClassHeader(string name, string docs, bool derive)
             return codeText;
         }
 
-        static string createConsumerSpecific()
+        static string createClassConstructors(string name)
         {
-            return
-@"        /// <summary>
-        ///     Initialize a new empty <see cref=""ConsumerConfig"" /> instance.
+            var codeText = "\n";
+            codeText += $@"        /// <summary>
+        ///     Initialize a new empty <see cref=""{name}"" /> instance.
         /// </summary>
-        public ConsumerConfig() {}
+        public {name}() {{ }}
 
         /// <summary>
-        ///     Initialize a new <see cref=""ConsumerConfig"" /> instance based on
+        ///     Initialize a new <see cref=""{name}"" /> instance wrapping
         ///     an existing <see cref=""ClientConfig"" /> instance.
         /// </summary>
-        public ConsumerConfig(ClientConfig config) { this.properties = new Dictionary<string, string>(config.ToDictionary(a => a.Key, a => a.Value)); }
+        /// <summary>
+        ///     <para>
+        ///     This will change the values ""in-place"" i.e. the original will be modified
+        ///     </para>
+        /// </summary>
+        public {name}(ClientConfig config) : base(config) {{ }}
 
         /// <summary>
-        ///     Initialize a new <see cref=""ConsumerConfig"" /> instance based on
+        ///     Initialize a new <see cref=""{name}"" /> instance wrapping
         ///     an existing key/value pair collection.
+        ///     <para>
+        ///     This will change the values ""in-place"" i.e. the original will be modified

```

Comment:
Maybe:
'Note: operations on this class will modify the provided collection'

Reply:
consolidated versions of the wording